packages/botonic-react/src/components/carousel.jsx

```diff
@@ -35,14 +57,81 @@ export const Carousel = props => {
     ...{ enable: true, autoHide: true },
     ...getThemeProperty('scrollbar'),
   }
+  const [hasLeftArrow, setLeftArrow] = useState(false)
+  const [hasRightArrow, setRightArrow] = useState(true)
+  const carouselRef = useRef(null)
+  const CustomCarouselLeftArrow = getThemeProperty(
+    'carousel.arrow.left',
+    undefined
+  )
+  const CustomCarouselRightArrow = getThemeProperty(
+    'carousel.arrow.right',
+    undefined
+  )
+  const enableCarouselArrows = getThemeProperty('carousel.enableArrows', true)
+
+  const scrollCarouselBy = value => {
+    carouselRef.current.scrollBy({
+      left: value,
+      behavior: 'smooth',
+    })
+  }
+
+  const setArrowsVisibility = event => {
+    const carousel = event.currentTarget
+    carousel.scrollLeft == 0 ? setLeftArrow(false) : setLeftArrow(true)
+    carousel.scrollLeft == carousel.scrollWidth - carousel.offsetWidth
```

Comment:
here you could do the same, but since it's a bit more complex maybe you can create a temporary variable to document what it does. St like (please rename the temp variable as I don't fully follow the code)
```
const carouselFits =     carousel.scrollLeft == carousel.scrollWidth - carousel.offsetWidth
setRightArrow(carouselFits)
```